src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -74,56 +74,56 @@ internal unsafe struct ManagedObjectWrapper
 
             internal volatile CreateComInterfaceFlagsEx Flags;
             const ulong ComRefCountMask = 0x000000007fffffffUL;
-            static ulong GetComCount(ulong c)
+            static uint GetComCount(ulong c)
             {
-                return c & ComRefCountMask;
+                return (uint)(c & ComRefCountMask);
             }
 
-            public ulong AddRef()
+            public uint AddRef()
             {
                 return GetComCount((ulong)Interlocked.Increment(ref RefCount));
             }
 
-            public ulong Release()
+            public uint Release()
             {
                 if (GetComCount((ulong)RefCount) == 0)
                 {
                     Debug.Fail("Over release of MOW - COM");
-                    return unchecked((ulong)-1);
+                    return unchecked((uint)-1);
                 }
 
                 return GetComCount((ulong)Interlocked.Decrement(ref RefCount));
             }
 
-            public unsafe int QueryInterface(ref Guid riid, out IntPtr ppvObject)
+            public unsafe int QueryInterface(Guid* riid, IntPtr* ppvObject)
```

Comment:
You can still use `ref`, `out` or `in` in the managed methods. It would look better than pointers.

To convert unmanaged pointer to one of these, you just need to add `*`. For example:

`wrapper->QueryInterface(in *guid, out *returnValue)`.